usecases/scenario_testrun_usecase.go

```diff
@@ -13,46 +13,41 @@ import (
 )
 
 type ScenarioTestRunUsecase struct {
-	transactionFactory             executor_factory.TransactionFactory
-	executorFactory                executor_factory.ExecutorFactory
-	scenarioPublicationsRepository repositories.ScenarioPublicationRepository
-	enforceSecurity                security.EnforceSecurityScenario
-	clientDbIndexEditor            clientDbIndexEditor
-	repository                     repositories.ScenarioTestRunRepository
+	transactionFactory  executor_factory.TransactionFactory
+	executorFactory     executor_factory.ExecutorFactory
+	enforceSecurity     security.EnforceSecurityTestRun
+	clientDbIndexEditor clientDbIndexEditor
+	repository          repositories.ScenarioTestRunRepository
+	scenarioRepository  ScenarioUsecaseRepository
 }
 
 func (usecase *ScenarioTestRunUsecase) ActivateScenarioTestRun(ctx context.Context,
 	organizationId string,
 	input models.ScenarioTestRunInput,
 ) (models.ScenarioTestRun, error) {
+	if err := usecase.enforceSecurity.CreateTestRun(organizationId); err != nil {
+		return models.ScenarioTestRun{}, err
+	}
 	exec := usecase.executorFactory.NewExecutor()
 	// we should not have any existing testrun for this scenario
-	existingTestrun, err := usecase.repository.GetByScenarioIterationID(ctx, exec, input.ScenarioIterationId)
+	existingTestrun, err := usecase.repository.GetTestRunByScenarioIterationID(ctx, exec, input.ScenarioIterationId)
```

Comment:
just thinking about this again: considering how the repository method is implemented, you have no guarantee here that _if an existing test run is found_, it's the most recent one (considering the condition you then apply line 38).
The safest thing would be to also pass a filter on "up" status test runs to the repo.

Reply:
good point.. which means there are two choices
- the one is about to apply what you just suggested, which means the method name should change (something like `GetActiveTestRunByScenarioID` )
- the other choice raise another question: is it possible to have many testruns for a given scenario_iteration_id ? if it is, then we could return all the testruns for a given scenario_iteration_id and check if there is any which has the status `up` ; in which case the method name would be `GetListTestrunScenarioByIterationID` 
what do you think ?

Reply:
Yes, that is correct.
Both work for me, so go for the one that seems simplest to you.
For what it's worth: we'll need some sort of "list test runs" repo method anyway soon.